### DISCUSSION TITLE
Pybricks cannot connect to Prime hub on Safari / Chrom / Edge / bluefy

### DISCUSSION THREAD

**Discussion by SVD81Tech** [pybricks/support#1902] [Q&A]
while trying to connect to the spike prime from the Chrome / safari / edge  browswe it given error that the web bluetooth is not supported on the browsers installed on ipad. 
tried using bluefy browser and it throws below exception 

Nv@https://code.pybricks.com/static/js/main.1bf60dfd.js:2:511695
h@https://code.pybricks.com/static/js/main.1bf60dfd.js:2:1401808
l@https://code.pybricks.com/static/js/main.1bf60dfd.js:2:1402282
@https://code.pybricks.com/static/js/main.1bf60dfd.js:2:1396510

how do we connect to the spike prime on chrome browser version on ipad?

**Comment by dlech** [comment]
We don't officially support Pybricks Code on iPad. Apple refuses to allow using Bluetooth from a web browser, so major browsers don't work. There are a number of modified browsers that claim to support WebBluetooth, but they have not kept up with the evolving standards. But there is one browser that is https://github.com/daphtdazz/WebBLE/issues/42 that might work (https://apps.apple.com/gb/app/webble/id1193531073)

Edit: That other browser seems to work in terms of Bluetooth but it has other usability issues so we don't recommend it.

**Reply by robtorok** [reply]
Any changes or possible workarounds for this? I coach a FIRST LEGO League team that's keen to go all in on Pybricks to program our robots this year, but it turns out that most of the team have iPads. We didn't realise this was an issues until after we'd bought a season licence. WebBLE allows up to do some programming, but we can't work out how to upload/download programs or even just copy & paste from the editor into say a google docs file.

Any suggestions?!?

**Reply by laurensvalk** [reply]
Hi Rob - nice to see you here. Unfortunately iPads are not supported due to platform limitations.

We do indicate the lack of iPad support [here](https://pybricks.com/learn/getting-started/what-do-you-need/#choosing-a-device-for-programming) but I understand it is easy to miss. We would be happy to refund your license if you decide not to use Pybricks because of this (please email us at team@pybricks.com in that case).
